plugins/source/aws/resources/services/sns/types.go

```diff
@@ -0,0 +1,54 @@
+package sns
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/sns/types"
+)
+
+// Amazon SNS subscription.
+type Subscription struct {
+	types.Subscription
+	// True if the subscription confirmation request was authenticated.
+	ConfirmationWasAuthenticated *bool
+	// The JSON serialization of the subscription's delivery policy.
+	DeliveryPolicy *string
+	// The JSON serialization of the effective delivery policy that takes into account the topic delivery policy and account system defaults.
+	EffectiveDeliveryPolicy *string
+	// The filter policy JSON that is assigned to the subscription.
+	FilterPolicy *string
+	// True if the subscription hasn't been confirmed.
+	PendingConfirmation *bool
+	// True if raw message delivery is enabled for the subscription.
+	RawMessageDelivery *bool
+	// When specified, sends undeliverable messages to the specified Amazon SQS dead-letter queue.
+	RedrivePolicy *string
+	// The ARN of the IAM role that has permission to write to the Kinesis Data Firehose delivery stream and has Amazon SNS listed as a trusted entity.
+	SubscriptionRoleArn *string
+}
+
+// Amazon SNS topic.
```

Comment:
Interesting! Why couldn't we use the SDK-provided `Topic` here?

Reply:
SDK provides most of the data in the form of `map[string]string`, see https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/sns#GetTopicAttributesOutput